server/src/parser.rs

```diff
@@ -0,0 +1,698 @@
+/* Copyright 2022 The ModelarDB Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//! Methods for tokenizing and parsing SQL commands. They are tokenized and
+//! parsed using [sqlparser] as it is already used by Apache Arrow DataFusion.
+//!
+//! [sqlparser]: https://crates.io/crates/sqlparser
+
+use datafusion::arrow::datatypes::{ArrowPrimitiveType, Field, Schema};
+use datafusion::common::DataFusionError;
+use datafusion_sql::planner;
+use sqlparser::ast::{
+    ColumnDef, ColumnOption, ColumnOptionDef, DataType, HiveDistributionStyle, HiveFormat, Ident,
+    ObjectName, Statement, TimezoneInfo,
+};
+use sqlparser::dialect::{Dialect, GenericDialect};
+use sqlparser::keywords::Keyword;
+use sqlparser::parser::{Parser, ParserError};
+use sqlparser::tokenizer::Token;
+
+use crate::metadata::{model_table_metadata::ModelTableMetadata, MetadataManager};
+use crate::models::ErrorBound;
+use crate::types::ArrowTimestamp;
+
+/// Constant specifying that a model table should be created.
+pub const CREATE_MODEL_TABLE_ENGINE: &str = "ModelTable";
+
+/// SQL dialect that extends `sqlparsers's` [`GenericDialect`] with support for
+/// parsing CREATE MODEL TABLE table_name DDL commands.
+#[derive(Debug)]
+struct ModelarDbDialect {
```

Comment:
Changed `ModelarDBError` to `ModelarDbError` as the [Rust Style Guide](https://github.com/rust-dev-tools/fmt-rfcs/blob/master/guide/advice.md) states that _"Where acronyms are used in the above cases, only capitalize the first letter, e.g., HttpRequest."_ and the [Rust API Guidelines](https://rust-lang.github.io/api-guidelines/naming.html) states that _"In UpperCamelCase, acronyms and contractions of compound words count as one word: use Uuid rather than UUID, Usize rather than USize or Stdin rather than StdIn."_.